source/common/ssl/context_manager_impl.cc

```diff
@@ -47,7 +139,7 @@ size_t ContextManagerImpl::daysUntilFirstCertExpires() {
 }
 
 void ContextManagerImpl::iterateContexts(std::function<void(Context&)> callback) {
-  std::unique_lock<std::mutex> lock(contexts_lock_);
+  std::unique_lock<std::shared_timed_mutex> lock(contexts_lock_);
```

Comment:
Why isn't this a shared lock? Is it because the callback takes a non-const Context? I think you could fix that and make the iterate callback take a const Context.

Reply:
I asked to make it non-shared because of the non-const callback param.  If all callbacks passed to this can use a const Context& then the type should be changed and this can be a shared lock.

Reply:
Done.